### PR TITLE
[webhook] Add admission check webhook

### LINKED ISSUE #1218: Add AdmissionCheck webhook.
**What would you like to be added**:
A webhook for AdmissionChecks
**Why is this needed**:

Make some of it's fields immutable  (`ControllerName`).
https://github.com/kubernetes-sigs/kueue/pull/1154/files#r1353671911

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add admission check webhook. 

#### Which issue(s) this PR fixes:

Fixes #1218

#### Does this PR introduce a user-facing change?

```release-note
Validate AdmissionCheck objects
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
Validate AdmissionCheck objects
```

File: pkg/webhooks/admissioncheck_webhook.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *AdmissionCheckWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	return nil, nil
+}
+
+func validateAdmissionCheck(ac *kueue.AdmissionCheck) field.ErrorList {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+	if len(ac.Spec.ControllerName) == 0 {
+		allErrs = append(allErrs, field.Invalid(specPath.Child("controllerName"), ac.Spec.ControllerName, utilvalidation.EmptyError()))
```
Comment (alculquicondor) [inline pkg/webhooks/admissioncheck_webhook.go:81]: you can use `field.Required` instead

**Review by alculquicondor — COMMENTED** [review]
/approve
/hold for nit

**Review by tenzen-y — COMMENTED** [review]
lgtm

File: pkg/webhooks/admissioncheck_webhook.go (hunk 1 above)
Comment (trasc) [reply pkg/webhooks/admissioncheck_webhook.go]: done

**Review by tenzen-y — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
NONE
```
Because it's a continuation to #1045

**Pull request merged**